### PR TITLE
Remove // TODO(mrhohn): Remove this function when kube-proxy is run as a DaemonSet by default

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**

/kind cleanup
/priority backlog
/cc @mrhohn

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref:https://github.com/kubernetes/kubernetes/issues/86763

**Special notes for your reviewer**:

In my mind, Kube-proxy is DaemonSet. So I do this.

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

File: test/e2e/framework/nodes_util.go [hunk 1]
```diff
@@ -191,11 +165,34 @@ func NodeUpgrade(f *Framework, v string, img string) error {
 	return waitForNodesReadyAfterUpgrade(f)
 }
 
+// MasterUpgradeGCEWithKubeProxyDaemonSet upgrades master node on GCE with enabling/disabling the daemon set of kube-proxy.
```
Comment (tanjunchen) [inline test/e2e/framework/nodes_util.go]: forget change the comment,i will update it after review by others

**Comment by MrHohn** [comment]
@tanjunchen Unfortunately kube-proxy doesn't currently run as daemonset by default in gce test suite:
https://github.com/kubernetes/kubernetes/blob/06abedb06314b02a3b2605ea86bde25b2bac11a8/cluster/gce/config-test.sh#L505

**Comment by tanjunchen** [comment]
> @tanjunchen Unfortunately kube-proxy doesn't currently run as daemonset by default in gce test suite:
> https://github.com/kubernetes/kubernetes/blob/06abedb06314b02a3b2605ea86bde25b2bac11a8/cluster/gce/config-test.sh#L505

get it

**Comment by oomichi** [comment]
Then we should not remove the function at this time according to the TODO message, right?

**Comment by tanjunchen** [comment]
> Then we should not remove the function at this time according to the TODO message, right?

yes
/hold

**Comment by alejandrox1** [comment]
@MrHohn is there an open issue or anything to track the work for running kube-proxy as a daemonset?
My guess for the test suite is that it would use the cluster addons https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/kube-proxy/kube-proxy-ds.yaml :thinking:

**Comment by MrHohn** [comment]
@alejandrox1 Yep, https://github.com/kubernetes/kubernetes/issues/23225 is the tracking issue.

**Pull request closed without merging**